app/src/main/java/com/fbada/projectone/Car.kt

```diff
@@ -0,0 +1,349 @@
+    /**
+     * @property seats number of seats
+     * @property location pickup location
+     */
+    var seats: Int = 8
+    val location: String = "Manchester"
+    /**
+     * @property price base price becomes 100.00
+     */
+    override var price: Double = 100.00
+
+    /**
+     * Start test drive
+     */
+    fun testDrive() {
+        println("Lets take this for a spin!")
+    }
+
+    /**
+     * Opens the boot
+     */
+    open fun openBoot() {
+        println("boot is open")
+    }
+
+
+}
+
+/**
+ * Subclass Alpha Model type car.  Extends BMW class.
+ * @property dashboard Big or small dashboard type
+ * @property satNav Sat Nav available
+ * @property ecoRating Eco Rating Score
+ */
+class Alpha(private val dashboard: String, val satNav: Boolean, val ecoRating: Int) : BMW("Diesel", "Manual") {
+    /**
+     * @property price Base price plus eco rating multiplied by 2.
+     */
+    override var price: Double = super.price + (ecoRating * 2)
+    /**
+     * Navigation function which prints a message dependent on whether the dashboard is big or small.
+     */
+    override fun navigate() {
+        if (dashboard == "big") {
+            println("Alpha is so easy to navigate")
+        } else {
+            println("Alpha map is too small")
+        }
+    }
+}
+
+/**
+ * Subclass Beta Model type car.  Extends BMW class.
+ * @property insurance Insurance available
+ * @property dashboard Big or small dashboard type
+ * @property powerSteering Power steering available
+ */
+class Beta(private val insurance: Boolean, private val dashboard: String, private val powerSteering: Boolean) :
+    BMW("Petrol", "Manual") {
+    /**
+     * Navigation function which prints a message dependent on whether the dashboard is big or small.
+     */
+    override fun navigate() {
+        if (insurance) {
+            if (dashboard == "big") {
+                price * 40
+                println("Beta is so easy to navigate")
+            } else {
+                price / 3
+                println("Beta map is too small")
+            }
+        } else {
+            println("This car needs insurance!")
+        }
+    }
+}
+/**
+ * Subclass Omega Model type car.  Extends BMW class.
+ * @property windowSize Size of windows
+ * @property dashboard Big or small dashboard type
+ * @property ecoRating Eco Rating Score
+ */
+class Omega(private val windowSize: Int, private val dashboard: String, val ecoRating: Int) : BMW("Petrol", "Manual") {
+    /**
+     * Navigation function which prints a message dependent on whether the dashboard is big or small.
+     */
+    override fun navigate() {
+        if (dashboard == "big") {
+            price * 20
+            println("Omega is so easy to navigate")
+        } else {
+            price / 13
+            println("Omega map is too small")
+        }
+    }
+}
+
+/**
+ * Subclass Mayback Model type car.  Extends Mercedes class.
+ */
+class Mayback : Mercedes(4, 3) {
```

Comment:
**Minor**: I think that's Maybach 😄 
4 and 3 is a bit hard to read(kinda [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))), especially during review. Usually it's helpfull to specify the param explicitly, by using  [named arguments](https://kotlinlang.org/docs/functions.html#named-arguments).  
In this case you would end up having something like:
`Mercedes(doors = 2, owners = 2)`
Initially, named arguments where introduced for a bit diff reasons, but using them this way will increase readability.
Bear in mind IDEA is smart enough to highlight those params even without specifying them directly, but that's not the case for "raw" text.

Reply:
Whoops- changed to the actual model, rather than the one I made up! XD

Changed all changed all constructors to include named arguments.